csharp/test/Ice/ami/AllTests.cs

```diff
@@ -55,10 +55,10 @@ public bool SentSynchronously
             }
         }
 
-        public void Report(bool sentSynchronously)
+        public void Report(bool value)
         {
-            SentSynchronously = sentSynchronously;
-            Sent = true;
+            _sentSynchronously = value;
+            _sent = true;
```

Comment:
Not familiar with this specific test, but it does seem a little suspicious not using the mutex here.